python/ray/dataframe/dataframe.py

```diff
@@ -555,16 +621,188 @@ def expanding(self, min_periods=1, freq=None, center=False, axis=0):
         raise NotImplementedError("Not Yet implemented.")
 
     def ffill(self, axis=None, inplace=False, limit=None, downcast=None):
-        raise NotImplementedError("Not Yet implemented.")
+        new_df = self.fillna(
+            method='ffill', axis=axis, limit=limit, downcast=downcast
+        )
+        if inplace:
+            self._df = new_df._df
+            self.columns = new_df.columns
+        else:
+            return new_df
 
     def fillna(self, value=None, method=None, axis=None, inplace=False,
                limit=None, downcast=None, **kwargs):
-        raise NotImplementedError("Not Yet implemented.")
+        if isinstance(value, (list, tuple)):
```

Comment:
We need some docs here.

Also, this one is really long. Let's go over it together.